### PR TITLE
Adding Improvements to Knuckledusters

### PR DESCRIPTION
**Pull request by TheKillfish**
## What Does This PR Do
Adds a few improvements to Knuckledusters as suggested by Balance Team and admins that tested it, with variations made by myself to try and maintain some balance/niches between the different types of knuckleduster. The changes are as follows;
- Normal and Engraved Dusters deal 10 damage, Syndicate deals 15.
- Removed their ability to break bones but retained their organ damage capabilities (this also removes internal bleeding on Slime People, as well as the health threshold for dealing organ damage).
- Increased cost of Syndie Dusters from 5 TC to 10 TC (May be increased further still to 15 TC if Balance feels it should).
- Made it sound like a normal punch.
- Gave the Engraved Dusters the same resistance flags as the Rapier, since steal objectives should be hard to destroy.

Todo List:
- [x] - Figure out randomising punch sound and reducing volume so it sounds even more normal, alternatively figuring out how to call the `soundin` proc for punch sounds instead of using `hitsound`. Currently just uses one of the four punch sounds, and is quite notably louder than normal punching.
- [x] - Figure out how to apply `ABSTRACT` examine info (that being a lack of examine info) without using the actual flag. This is so you can't examine to see if someone is holding it (suggested by marm). Problem with using the flag is manyfold but to narrow it down to the main issues; the flag prevents you from picking up, throwing or putting items with the flag in storage.

## Why It's Good For The Game
Improvements are always good, and the previous dusters were lacking in some areas. While I find the recommended removal of bonebreaking a shame and don't fully understand why, the other reasons given to me were understandable.

## Testing
Changed code, punched some Skrell

## Changelog
:cl:
tweak: Increased Knuckleduster damage, removed it's ability to break bones of boned species and internal bleeding on non-boned species, made it sound like normal punching, made Syndicate Dusters a bit more expensive to buy
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/knuckledusters.dm [hunk 1]
```diff
@@ -37,18 +36,11 @@
...
-		else
-			punched.fracture()
-		return
-
 	if(!length(punched.internal_organs))
 		return
 
 	var/obj/item/organ/internal/squishy = pick(punched.internal_organs)
-	if(gripped && elite && target.health < 90)
+	if(gripped && elite)
```
Comment (DGamerL) [inline code/game/objects/items/weapons/knuckledusters.dm:43]: Mention this somewhere in the PR

**Review by DGamerL — COMMENTED** [review]
Please make your changelog a bit more descriptive

File: code/game/objects/items/weapons/knuckledusters.dm [hunk 2]
```diff
@@ -71,16 +62,15 @@
 	force = 10
 	throwforce = 5
 	origin_tech = "combat=3"
-	resistance_flags = FIRE_PROOF | ACID_PROOF
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF // Copied from Rapier, I figure this should be made hard-to-destroy as well
```
Comment (Henri215) [inline code/game/objects/items/weapons/knuckledusters.dm:65]:
```suggestion
	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF // Steal objectives shouldnt be easy to destroy.
```

**Comment by TheKillfish** [comment]
Issues I had have been solved, thanks to advice from DGamer and Contra. Ready for undrafting, I feel.

File: code/game/objects/items/weapons/knuckledusters.dm [hunk 3]
```diff
@@ -4,65 +4,58 @@
...
-	var/robust = 10
 	/// How much organ damage can the weapon do?
 	var/trauma = 5
 
 /obj/item/melee/knuckleduster/attack_self(mob/user)
 	if(!gripped)
 		gripped = TRUE
 		to_chat(user, "You tighten your grip on [src], ensuring you won't drop it.")
-		flags |= NODROP
+		flags |= NODROP | ABSTRACT
```
Comment (DGamerL) [inline code/game/objects/items/weapons/knuckledusters.dm:26]:
```suggestion
		flags |= (NODROP | ABSTRACT)
```

File: code/game/objects/items/weapons/knuckledusters.dm [hunk 4]
```diff
@@ -4,65 +4,58 @@
...
 	else
 		gripped = FALSE
 		to_chat(user, "You relax your grip on [src].")
-		flags &= ~NODROP
-		return
+		flags &= ~(NODROP | ABSTRACT)
+
+/obj/item/melee/knuckleduster/attack/(mob/living/user)
+	hitsound = pick('sound/weapons/punch1.ogg','sound/weapons/punch2.ogg','sound/weapons/punch3.ogg','sound/weapons/punch4.ogg')
```
Comment (DGamerL) [inline code/game/objects/items/weapons/knuckledusters.dm:33]:
```suggestion
	hitsound = pick('sound/weapons/punch1.ogg', 'sound/weapons/punch2.ogg', 'sound/weapons/punch3.ogg', 
'sound/weapons/punch4.ogg')
```

**Review by S34NW — APPROVED** [review]

**Pull request merged**